modules/mini-games/tightspot.lua

```diff
@@ -187,143 +256,124 @@ local function player_join_game(player, at_player)
             markets[#markets + 1] = position
-        end
-        local ent = variables["surface"].create_entity {name = name, position = position, force = force}
-        if name == "steel-chest" then
+        elseif name == "steel-chest" then
             chests[#chests + 1] = {ent, player.name}
         end
         ent.minable = minable
     end
 
+    -- Find the center for the players island
     centers[player.name] = {
         x = level.center.x + player_offset,
         y = level.center.y
     }
 
-    variables.walls[player.name] = {}
-    local wal = variables["surface"].find_entities_filtered {name = "stone-wall", area = area}
-    for i, wall in ipairs(wal) do
+    -- Find all the walls on the island
+    local wall_data = {}
+    variables.walls[player.name] = wall_data
+    local walls_found = variables["surface"].find_entities_filtered{ name = "stone-wall", area = area }
+    for i, wall in ipairs(walls_found) do
         local p = wall.position
-        variables.walls[player.name][p.x .. "," .. p.y] = true
+        wall_data[p.x .. "," .. p.y] = true
     end
-    player.teleport(centers[player.name], level.surface)
-end
 
-local function level_save()
-    local level = variables.level
-    local tiles = save["tiles"]
-    for x = level.area[1][1], level.area[2][1] do
-        for y = level.area[1][2], level.area[2][2] do
-            local tile = variables["surface"].get_tile(x, y)
-            tiles[#tiles + 1] = {
-                name = tile.name,
-                position = tile.position
-            }
-        end
-    end
-
-    save.entities = variables["surface"].find_entities_filtered {area = level.area}
-    for i, entity in ipairs(save.entities) do
-        local name = entity.name
-        if name ~= "character" then
-            save.entities[i] = {name, entity.position, entity.force, entity.minable}
-        else
-            if i == #save.entities then
-                save.entities[i] = nil
-            else
-                local ent = save.entities[#save.entities]
-                save.entities[i] = {ent.name, ent.position, ent.force, ent.minable}
-                save.entities[#save.entities] = nil
-            end
-        end
-    end
+    -- Teleport the player to their island
+    player.teleport(centers[player.name], level.surface)
 end
 
-local function market_setup()
-    for i, pos in ipairs(markets) do
-        local market = variables["surface"].find_entity("market", pos)
-        market.clear_market_items()
-        for _, item in ipairs(variables.level.items) do
-            local offer = {price = {{"coin", tightspot_prices[item]}}, offer = {type = "give-item", item = item}}
-            market.add_market_item(offer)
-        end
-    end
-end
+--- Called by mine game module to start this minigame
 local function start(args)
     variables["level"] = {}
     variables["surface"] = {}
     variables["walls"] = {}
+
+    -- Setup all level related variables
     local level_index = args[1]
```

Comment:
this shood be tonumber(args[1]) as /start will other wise not work

Reply:
This line was not chaged, so this was an error with the src. Will add the correction.